appinventor/codelabs/mole-mash/index.html

```diff
@@ -0,0 +1,242 @@
+            <li>Explain what the TimerInterval property of the Clock component does</li>
+            <li>Describe what a sprite component is, and how it's used in your app</li>
+          </ul>
+        </aside>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Define a procedure to move the mole">
+        <p>Now it's time to code your app. To do that, we're going to need a new type of block: <i>procedures</i>. A procedure is a sequence of statements that you can refer to all at once as single command.</p>
+        <p>MoleMash requires 2 procedures. Go to <b>Blocks View</b>. Find the <b>Procedures</b> drawer under <b>Built-In</b>, and drag out a <b>to procedure</b> block. Change the label "procedure" to "MoveMole".</p>
+        <p>MoveMole is a procedure that moves the Mole sprite to a new random position on the Canvas. To do this, you will set the Mole's x and y positions to be a random fraction, between 0 and 1, of the difference between the size of the canvas and the size of the mole.</p>
+        <p>This is what the complete MoveMole procedure should look like:</p>
+        <img src="img/movemole.png">
+        <p>First, from the <b>Mole</b> drawer, drag a <code>set Mole.X to</code> block into the procedure.</p>
+        <img src="img/movemole1.png">
+        <p>From the <b>Math</b> drawer, drag a <i>multiplication</i> block into the <code>set Mole.X to</code> block.</p>
+        <img src="img/movemole2.png">
+        <p>From the <b>Math</b> drawer, drag a <code>random fraction</code> block into the first multiplication plug.</p>
+        <img src="img/movemole3.png">
+        <p>From the <b>Math</b> drawer, drag a <i>subtraction</i> block into the second multiplication plug.</p>
+        <img src="img/movemole4.png">
+        <p>From the <b>MyCanvas</b> drawer, drag a <code>MyCanvas.Width</code> block into the first subtraction plug.</p>
+        <img src="img/movemole5.png">
+        <p>From the <b>Mole</b> drawer, drag a <code>Mole.Width</code> block into the second subtraction plug.</p>
+        <img src="img/movemole6.png">
+        <h2>Challenges</h2>
+        <ol>
+          <li>Finish the MoveMole procedure by adding blocks to change the position of the Mole's y position.</li>
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Define a procedure to update the score">
+        <p>Our second prodecure will update the score of the game.</p>
+        <p>First, we need a variable to keep track of the score. Define a global variable named "score", and set the initial value to 0. This is what your block should look like:</p>
+        <img src="img/score_variable.png" style="max-width: 300px">
+        <p>Next, define a procedure named "UpdateScore". This procedure will change the text of the the label <b>ScoreLabel</b> to show the score of the game.</p>
+        <h2>Challenges</h2>
+        <ol>
+          <li>Try to code this procedure by yourself.</li>
```

Comment:
Instead of pasting the answer below, can you grab a screenshot of the different blocks involved (not hooked up together) and instruct the students to "Implement the `UpdateScore` procedure, which should make the **ScoreLabel** show the game score. The blocks below might be helpful."